.github/workflows/main.yml

```diff
@@ -551,36 +543,30 @@ jobs:
         name: prelude-docs.tar.xz
         path: prelude-docs.tar.xz
 
-  benchmark-opencl:
-    runs-on: ubuntu-22.04
+  benchmark-A100:
+    runs-on: A100
     needs: [build-linux-nix]
-
+    if: github.repository == 'diku-dk/futhark'
     steps:
     - uses: actions/checkout@v3
       with:
         submodules: recursive
         fetch-depth: 0
-
-    - name: Install dependencies
-      run: |
-        sudo apt-get update
-        sudo apt-get install -y opencl-headers nvidia-cuda-dev nvidia-opencl-dev
-
     - uses: actions/download-artifact@v3
       with:
         name: futhark-nightly-linux-x86_64.tar.xz
-
-    - name: Install from nightly tarball
+    - name: Setup compiler
       run: |
         tar xvf futhark-nightly-linux-x86_64.tar.xz
-        make -C futhark-nightly-linux-x86_64/ install PREFIX=$HOME/.local
-        echo "$HOME/.local/bin" >> $GITHUB_PATH
-
+        echo "$PWD/futhark-nightly-linux-x86_64/bin" >> $GITHUB_PATH
     - run: |
         cd futhark-benchmarks && pwd && ./get-data.sh external-data.txt
-
-    - run: |
-        futhark test -C --no-terminal --backend=opencl  --pass-compiler-option=--Werror futhark-benchmarks
+    - name: Benchmark OpenCL
+      run: |
+        futhark bench futhark-benchmarks --no-terminal --backend=opencl --json futhark-opencl-A100-$GITHUB_SHA.json
+    - name: Benchmark CUDA
+      run: |
+        futhark bench futhark-benchmarks --no-terminal --backend=cuda --json futhark-cuda-A100-$GITHUB_SHA.json
```

Comment:
Now that we're actually doing this, we could just push directly to https://futhark-lang.org/benchmark-results/, right?

Reply:
Yes, that is the next step.